vortex-alp/src/compute.rs

```diff
@@ -61,3 +68,52 @@ impl SliceFn for ALPArray {
         .into_array())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use vortex::array::primitive::PrimitiveArray;
+    use vortex::compute::as_contiguous::AsContiguousFn;
+    use vortex::compute::scalar_at::scalar_at;
+    use vortex::validity::Validity;
+    use vortex::IntoArray;
+
+    use crate::ALPArray;
+
+    #[test]
+    fn test_as_contiguous() {
+        let values = vec![1.0, 2.0, 3.0];
+        let primitives = PrimitiveArray::from_vec(values, Validity::NonNullable);
+        let encoded = ALPArray::encode(primitives.into_array()).unwrap();
+        let alp = ALPArray::try_from(&encoded).unwrap();
+
+        let flat = alp.as_contiguous(&[encoded]).unwrap();
+
+        let a = scalar_at(&flat, 0)
+            .unwrap()
+            .value()
+            .as_pvalue()
+            .unwrap()
+            .unwrap();
+        let a: f64 = a.try_into().unwrap();
+
+        let b = scalar_at(&flat, 1)
+            .unwrap()
+            .value()
+            .as_pvalue()
+            .unwrap()
+            .unwrap();
+        let b: f64 = b.try_into().unwrap();
+
+        let c = scalar_at(&flat, 2)
+            .unwrap()
+            .value()
+            .as_pvalue()
+            .unwrap()
+            .unwrap();
+        let c: f64 = c.try_into().unwrap();
+
+        assert_eq!(a, 1.0);
+        assert_eq!(b, 2.0);
+        assert_eq!(c, 3.0);
```

Comment:
I'm not sure if there's a more concise way to get the `f64`s back out of the flattened thing

Reply:
Can have the test `-> VortexResult<()>` and then use `?`. I think you can also just try_into on the scalar without unwrapping as a value

Reply:
i ended up just going with the try_into() that rob pointed out below